backfiller/backfiller/main.py

```diff
@@ -529,14 +558,45 @@ def backfill(self):
 				hours_backfilled.labels(remote=self.node, channel=channel, quality=quality).inc()
 
 
+	def backfill_extras(self):
+		for dir in self.extras:
+			self.logger.info(f'Backfilling extra directory {dir}')
+
+			local_files = set(list_local_extras(self.base_dir, dir))
+			remote_files = set(list_remote_extras(self.node, dir))
+			missing_files = list(remote_files - local_files)
+
+			# randomise the order of the files to reduce the chance that
+			# multiple workers request the same file at the same time
+			random.shuffle(missing_files)
+
+			pool = gevent.pool.Pool(self.download_concurrency)
+			workers = []
+
+			for file in missing_files:
+				if self.stopping.is_set():
+					return
+				path = os.path.join(dir, file)
+				self.logger.info(f"Backfilling {path}")
+				workers.append(pool.spawn(
+					get_remote_extra(self.base_dir, self.node, path)
+				))
+
+		for worker in workers:
+			worker.get()
+
+		self.logger.info("Backfilled extras")
```

Comment:
I would include the number of files back filled to this info message - it is just len(missing_files)

Reply:
done